src/models/BibDetails.ts

```diff
@@ -0,0 +1,209 @@
+    const bibFieldValue = this.bib.holdings[fieldMapping.field]
+    return this.buildDetail(fieldMapping.label, bibFieldValue)
+  }
+
+  buildStandardDetail(fieldMapping: FieldMapping) {
+    const bibFieldValue = this.bib[fieldMapping.field]
+    return this.buildDetail(fieldMapping.label, bibFieldValue)
+  }
+
+  buildDetail(label, value) {
+    if (!value?.length) return null
+    return { label, value }
+  }
+
+  buildInternalLinkedDetail(fieldMapping: {
+    label: string
+    field: string
+  }): LinkedBibDetail {
+    const value = this.bib[fieldMapping.field]
+    if (!value?.length) return null
+    return {
+      link: "internal",
+      label: fieldMapping.label,
+      value: value.map((v: string) => {
+        const internalUrl = `/search?filters[${fieldMapping.field}][0]=${v}`
+        return { url: internalUrl, urlLabel: v }
+      }),
+    }
+  }
+
+  get extent(): BibDetail {
+    let modifiedExtent: string[]
+    const { extent, dimensions } = this.bib
+    const removeSemiColon = (extent) => [extent[0].replace(/\s*;\s*$/, "")]
+    const extentExists = extent && extent[0]
+    const dimensionsExists = dimensions && dimensions[0]
+    if (!extentExists && !dimensionsExists) return null
+    if (!extentExists && dimensionsExists) modifiedExtent = dimensions
+    if (extentExists && !dimensionsExists) {
+      modifiedExtent = removeSemiColon(extent)
+    }
+    if (extentExists && dimensionsExists) {
+      const parts = removeSemiColon(extent)
+      parts.push(dimensions[0])
+      modifiedExtent = [parts.join("; ")]
+    }
+    return {
+      label: "Description",
+      value: modifiedExtent,
+    }
+  }
+
+  get supplementaryContent(): LinkedBibDetail {
+    if (
+      !this.bib.supplementaryContent?.length ||
+      // This is not for a known edge case, just here as a safeguard to avoid
+      // broken a tags.
+      !this.bib.supplementaryContent[0]?.url
+    ) {
+      return null
+    }
+    const label = "Supplementary Content"
+    const values = this.bib.supplementaryContent.map((sc) => {
+      return {
+        url: sc.url,
+        urlLabel: sc.label,
+      }
+    })
+    return {
+      link: "external",
+      label,
+      value: values,
+    }
+  }
+
+  get topDetails() {
+    return [
+      { field: "titleDisplay", label: "Title" },
+      { field: "publicationStatement", label: "Published By" },
+      // external link
+      { field: "supplementaryContent", label: "Supplementary Content" },
+      // internal link
+      { field: "creatorLiteral", label: "Author" },
+    ]
+      .map((fieldMapping) => {
+        let detail
+        if (fieldMapping.field === "supplementaryContent")
+          detail = this.supplementaryContent
+        else if (fieldMapping.field === "creatorLiteral")
+          detail = this.buildInternalLinkedDetail(fieldMapping)
+        else detail = this.buildStandardDetail(fieldMapping)
+        if (!detail) return
+        return detail
+      })
+      .filter((f) => f)
```

Comment:
Out of curiosity, what are these filters doing?

Reply:
filtering out null values. If a bib doesn't have a supplementary content, for example, we don't want there to be a null value in the array of details.